Source/CarthageKit/Xcode.swift

```diff
@@ -188,6 +188,23 @@ public func xcodebuildTask(_ task: String, _ buildArguments: BuildArguments, env
 	return xcodebuildTask([task], buildArguments)
 }
 
+func shouldBuildXCFramework(deviceBuildSettings: BuildSettings, simulatorBuildSettings: BuildSettings) -> Result<Bool, CarthageError> {
```

Comment:
This PR has me pretty excited, so I pulled this branch to try it out.  In my limited testing, everything is working well, but I ran into strange errors with Xcode 11.7.  

This method sounds like what is needed, but I don't think this is called from anywhere.

Reply:
That fix makes total sense. I'll incorporate it when I get a chance to!

re: Xcode < 12, it sounds `--create-xcframework` should fail with an error like "you need to be on Xcode 12 or greater", right? I added an `.xcframeworkRequried` error in this PR, but didn't think to  incorporate it in the argument parser. Should be pretty easy to change though.

`shouldBuildXCFramework(deviceBuildSettings:simulatorBuildSettings:)` stuck around from the initial implementation of this PR, where Carthage would automatically make an xcframework if it determined that lipo would fail. Now that we're producing multi-platform xcframeworks, I'm not sure that it has any utility.

Reply:
@jeffctown https://github.com/Carthage/Carthage/pull/3071/commits/18991ec8348ec63be19318f967297e063257f200 has both of these incorporated, lmk if you run into any problems or have additional feedback!

I took a slightly different approach on static frameworks (using build settings instead of dwarfdump to save time), but I was able to reproduce and confirm locally. Thanks so much for reporting this :)

Reply:
@elliottwilliams Awesome!  Thanks so much for bringing these changes in.  Your approach makes complete sense.  🎉

I'll give it a test run this week and let you know how it goes.

Reply:
Just reporting back that the static frameworks fix is working like a charm for my project.  Thank you for this contribution @elliottwilliams !